nixos/modules/services/system/swapspace.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.swapspace;
+in {
+  ###### interface
+
+  options = {
+
+    services.swapspace = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to create swapfiles dynamically using the SwapSpace manager.
+          Files will be added and removed based on current memory usage.
+        '';
+      };
+
+      path = mkOption {
+        type = types.path;
+        default = "/var/lib/swap";
```

Comment:
Maybe we could add an option for drives.

Reply:
@SuperSandro2000 could you elaborate? Do you mean people choosing devices instead of paths? That seems really hard to implement vs having them choose a path?

Reply:
I have a swap drive to not murder my main drive and keep it longer. I don't think it is that hard tbh. I just told swap to use /dev/sdX and then it was done.

Reply:
ah - swapspace actually manages files, not partitions, so that's not possible.